spec/unit/node_spec.rb

```diff
@@ -784,6 +784,50 @@
         expect(node["passenger"]["root_path_2"]).to eql("passenger-4.0.57")
         expect(node[:passenger]["root_path_2"]).to eql("passenger-4.0.57")
       end
+
+      it "should deep merge array attributes defined as literals" do
+        node.default["attr_literal_hash"] = { "key" => { "inner" => "value" } }
+        expect(node[:attr_literal_hash][:key]).not_to be_nil
+        expect(node[:attr_literal_hash][:key]).to be_a_kind_of(Mash)
+        expect(node[:attr_literal_hash]["key"]).not_to be_nil
+        expect(node["attr_literal_hash"]["key"]).not_to be_nil
+        expect(node["attr_literal_hash"][:key]).not_to be_nil
+        expect(node[:attr_literal_hash][:key][:inner]).to eql("value")
+        expect(node[:attr_literal_hash]["key"][:inner]).to eql("value")
+        expect(node[:attr_literal_hash]["key"]["inner"]).to eql("value")
+        expect(node[:attr_literal_hash][:key]["inner"]).to eql("value")
+        expect(node["attr_literal_hash"][:key][:inner]).to eql("value")
+        expect(node["attr_literal_hash"]["key"][:inner]).to eql("value")
+        expect(node["attr_literal_hash"]["key"]["inner"]).to eql("value")
+        expect(node["attr_literal_hash"][:key]["inner"]).to eql("value")
+        expect(node["attr_literal_hash"][:key]["not_existing"]).to be_nil
+
+        node.default["attr_literal_array"] = [{ "key" => { "inner" => "value" } }]
+        expect(node[:attr_literal_array]).to be_a_kind_of(Array)
```

Comment:
^ this still seems to be wrong, it should be an ImmutableArray when you get it through the merged `node[:foo]` accessor and should be an AttrArray when going through the `node.default[:foo]` accessor.